drivers: clock control: stm32U5 set regulator voltage before clocks

FRASTM opened the pull request:
The regulator voltage should be set before the clocks are enabled
This is especially the case when the MSIS at 48MHz is selected as
SYSCLK.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/48081

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
drivers: clock control: stm32U5 set regulator voltage before clocks

The regulator voltage should be set before the clocks are enabled.
This is especially the case when the MSIS at 48MHz is selected as
SYSCLK.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
tests: drivers: clock_control rename stm32 ahb test configuration

rename to pll_msis_ahb_2_40

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
testing  `tests/drivers/clock_control/stm32_clock_configuration/stm32u5_core` on b_u585i_IOT02a disco kit :
```
drivers.stm32_clock_configuration.u5.sysclksrc_pll_msis_160 : PASSED
drivers.stm32_clock_configuration.u5.pll_msis_ahb_2_40 : PASSED
drivers.stm32_clock_configuration.u5.sysclksrc_pll_hsi_160 : PASSED
drivers.stm32_clock_configuration.u5.sysclksrc_pll_hsi_40 : PASSED
drivers.stm32_clock_configuration.u5.sysclksrc_msis_48 : PASSED
drivers.stm32_clock_configuration.u5.sysclksrc_msis_24 : PASSED
```

str4t0m commented on drivers/clock_control/clock_stm32_ll_u5.c:
```diff
@@ -561,9 +564,6 @@ int stm32_clock_control_init(const struct device *dev)
-	set_regu_voltage(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
-
 	/* Set flash latency */
 	/* If freq increases, set flash latency before any clock setting */
```
I believe the configuration of the flash latency, in case old < new, should be moved up as well.

FRASTM replied:
OK, to set flash latentcy before _any_ clock setting
(the refMan says to "Adjust number of wait states according new frequency" before the PLL config.)

FRASTM committed:
drivers: clock control: stm32U5 set flash latency before clocks

As mentionned here, setting flash latency before any clock setting,
and especially before the PLL.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

str4t0m reviewed (approved):
Thanks
